src/zappy_gui_src/Core/SFML/Interface/Minimap/Minimap.hpp

```diff
@@ -0,0 +1,96 @@
+/*
+** EPITECH PROJECT, 2022
+** zappy
+** File description:
+** Minimap
+*/
+
+/// \file src/zappy_gui_src/Core/SFML/Interface/Minimap.hpp
+
+#ifndef MINIMAP_HPP_
+#define MINIMAP_HPP_
+
+#include <vector>
+#include <tuple>
+#include <string>
+#include <SFML/Graphics.hpp>
+#include <memory>
+
+namespace gui {
+    /// \brief A class for the minimap.
+    class Minimap {
+        public:
+            /// \brief Constructor of the Minimap.
+            Minimap();
+
+            /// \brief Destructor of the Minimap.
+            ~Minimap() = default;
+
+            /// \brief Display the minimap.
+            void display();
+
+            /// \brief Add a player from the minimap.
+            /// \param name The name of the player.
+            /// \param position The position of the player.
+            inline void addPlayer(const std::string &name, const sf::Vector2f &position) {
+                _playerList.push_back(std::tuple(name, position));
+            };
+
+            /// \brief Remove a player from the minimap.
+            /// \param name The name of the player.
+            void removePlayer(const std::string &name);
+
+            /// \brief Set the map size.
+            /// \param mapSize The map size.
+            inline void setMapSize(const sf::Vector2f &size) {
+                _mapSize = size;
+                _updateConvexShape();
+            };
+
+            /// \brief Switch the minimap size.
```

Comment:
Switch the minimap size, cool but can u explain a bit more pls, why use it when should we use it, why this function exist, can u developp, same for all others functions, documentation is usefull when the description is complete, useless otherwise